### ISSUE TITLE
Nushell doesn't update SHLVL

### ISSUE THREAD

**Issue by upidapi** [nushell/nushell#14384]
### Describe the bug

Nushell doesn't increment SHLVL when you run a sub shell 

You can partially remedy this by adding this to your config.nu
```nushell
$env.SHLVL = $env | get -si SHLVL | default 0 | into int | $in + 1
```
But that also increments it when using exec

You can remedy that with the following. (There's probably some better way than this)
```nushell
alias _exec = exec
def exec [
    command: string
    --help (-h): string
] {
    $env.SHLVL -= 1

    if ($help | is-empty) {
        _exec $command
    } else {
        _exec -h $help
    }
}
```

### How to reproduce

```bash
echo $env.SHLVL # => 1

exec nu
echo $env.SHLVL # => 1

nu
# SHLVL should increment
echo $env.SHLVL # => 1
```

### Expected behavior

```bash
echo $env.SHLVL # => 1

exec nu
echo $env.SHLVL # => 1

nu
echo $env.SHLVL # => 2
```

### Configuration

| key                | value                                                             |
| ------------------ | ----------------------------------------------------------------- |
| version            | 0.99.1                                                            |
| major              | 0                                                                 |
| minor              | 99                                                                |
| patch              | 1                                                                 |
| branch             |                                                                   |
| commit_hash        |                                                                   |
| build_os           | linux-x86_64                                                      |
| build_target       | x86_64-unknown-linux-gnu                                          |
| rust_version       | rustc 1.82.0 (f6e511eec 2024-10-15) (built from a source tarball) |
| cargo_version      | cargo 1.82.0 (8f40fc59f 2024-08-21)                               |
| build_time         | 1980-01-01 00:00:00 +00:00                                        |
| build_rust_channel | release                                                           |
| allocator          | mimalloc                                                          |
| features           | default, sqlite, trash                                            |
| installed_plugins  |                                                                   |

**Comment by fdncred** [comment]
What's the point of SHLVL? What benefit is it to people if nushell updates it?

**Comment by upidapi** [comment]
And how many times i can call exit before i revert the shell to the base state

**Comment by devyn** [comment]
It would be a pretty easy thing to implement I think, so if someone really wants to it should be an easy first contribution

**Comment by NotTheDr01ds** [comment]
> What's the point of SHLVL? What benefit is it to people if nushell updates it?

It's often included in the prompt so that folks know when they are in a nested-shell.  It's included as a Starship feature, for instance.  I'd probably add it to mine to have a better indication of when I'm running in `./target/debug/nu` (etc.)

**Comment by NotTheDr01ds** [comment]
And yes, to @devyn 's point, probably an easy change.

Just check for the existence of a SHLVL environment variable:

* Exists and is int: increment by 1.
* Doesn't exist: Create and set to 1.

**Comment by fdncred** [comment]
I'm wondering how easy/hard this is since nushell stores env vars internally versus at the system level. If we actually have to make a system call to set a system env var, that might be slightly costly on startup. We may be able to get away with just incrementing the SHLVL in nushell's internal env vars and never decrementing it since each nu instance will have its own.

Example.
nu = SHLVL 1
nu nu = SHLVL 2 but when you exit the first one, it inherits the old one
nu nu nu = SHLVL 3 same as above, when you exit, it shows 2

I'm not sure if that would work but it may be worth a test.

**Comment by NotTheDr01ds** [comment]
> I also wonder if this could be done as a script in the config.nu.

Good point - Almost certainly.

**Comment by fdncred** [comment]
I added this to my env.nu
```nushell
def --env do_shlvl [] {
    $env.SHLVL = $env.SHLVL? | default 0 | into int | $in + 1
}
```
and then added this to my config.nu
```nushell
do_shlvl
```
Seems to work ok.

**Comment by NotTheDr01ds** [comment]
Definitely easier than what I was thinking (an ENV_CONVERSION for it ;-)).

As an alternative, just put your one-liner in `($nu.data_dir)/vendor/autoload/shlvl.nu`:

```nushell
# ($nu.data-dir | path join "vendor/autoload/shlvl.nu")
$env.SHLVL = $env.SHLVL? | default 0 | into int | $in + 1
```

**Comment by NotTheDr01ds** [comment]
Ouch - I should read the OP better.

Me:

> I can't think of a reason why that wouldn't work.

OP:

> But that also increments it when using exec

**Comment by upidapi** [comment]
it actually does work for exec from other shells
![image](https://github.com/user-attachments/assets/fbe6da78-9bf9-4547-96a1-3558751f9011)
I don't know exactly why but id guess that most shells simply decrement SHLVL in exec

**Comment by upidapi** [comment]
"6 hours of debugging can save you from 5 minutes of reading the documentation"

**Comment by WindSoilder** [comment]
The issue is resolved by #14404
